### PR TITLE
refactor: Steps full token

### PR DESCRIPTION
**Pull request by MadCcc**
- refactor: steps full token
- refactor: progress dot
- refactor: no hard code

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   -        |
| 🇨🇳 Chinese |    -       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/steps/style/index.tsx [hunk 1]
```diff
@@ -131,17 +122,15 @@ const genStepsItemStyle: GenerateStyle<StepsToken, CSSObject> = token => {
...
+      marginInlineStart: 0,
+      marginInlineEnd: token.marginXS,
       fontSize: token.stepsIconFontSize,
       fontFamily: token.fontFamily,
       lineHeight: `${token.stepsIconSize}px`,
       textAlign: 'center',
-      border: `${token.controlLineWidth}px ${token.controlLineType} ${new TinyColor('#000')
-        .setAlpha(0.25)
-        .toRgbString()}`, // FIXME: hardcode in v4
+      border: `${token.controlLineWidth}px ${token.controlLineType} ${token.colorTextDisabled}`,
```
Comment (zombieJ) [inline components/steps/style/index.tsx:133]: colorPlaceholder?

Comment (MadCcc) [reply components/steps/style/index.tsx:133]: 这个和文字一个颜色

File: components/steps/style/index.tsx [hunk 2]
```diff
@@ -1,46 +1,37 @@
...
   stepsIconFontSize: number;
-  stepsIconMargin: string;
   stepsTitleLineHeight: number;
   stepsSmallIconSize: number;
-  stepsSmallIconMargin: string;
   stepsDotSize: number;
-  stepsDotTop: number;
   stepsCurrentDotSize: number;
-  stepsDescriptionMaxWidth: number;
+  descriptionWidth: number;
```
Comment (zombieJ) [inline components/steps/style/index.tsx:33]: StepsToken 里最好还是带着 step 前缀，这样不容易起冲突

File: components/steps/style/index.tsx [hunk 3]
```diff
@@ -318,53 +306,52 @@ const genStepsStyle: GenerateStyle<StepsToken, CSSObject> = token => {
...
-  });
+    const stepsToken = mergeToken<StepsToken>(token, {
+      // Steps variable default.less
+      processTailColor,
+      stepsNavArrowColor: new TinyColor('#000').setAlpha(0.25).toRgbString(), //  fade(@black, 25%),
+      stepsIconSize,
+      stepsIconCustomSize: stepsIconSize,
+      stepsIconCustomTop: 0,
+      stepsIconCustomFontSize: token.fontSizeIcon * 2,
+      stepsIconTop: -0.5, // magic for ui experience
```
Comment (zombieJ) [inline components/steps/style/index.tsx:324]: 审查元素看这个值一直不生效可以删掉不

File: components/steps/style/index.tsx [hunk 4]
```diff
@@ -318,53 +306,52 @@ const genStepsStyle: GenerateStyle<StepsToken, CSSObject> = token => {
...
+      finishIconColor: processIconColor,
+      finishTitleColor: token.colorText,
+      finishDescriptionColor: token.colorTextSecondary,
+      finishTailColor: token.colorPrimary,
+      errorIconColor: token.colorError,
+      errorTitleColor: token.colorError,
+      errorDescriptionColor: token.colorError,
+      errorTailColor: processTailColor,
+      stepsNavActiveColor: token.colorPrimary,
+      stepsProgressSize: 40,
```
Comment (zombieJ) [inline components/steps/style/index.tsx:349]: controlHeightLG

File: components/steps/style/index.tsx (hunk 2 above)
Comment (MadCcc) [reply components/steps/style/index.tsx:33]: 这个移到 ComponentToken 里了。。忘记删了

File: components/steps/style/index.tsx (hunk 3 above)
Comment (MadCcc) [reply components/steps/style/index.tsx:324]: 有些细微差别，所以还是留着了

File: components/steps/style/index.tsx (hunk 2 above)
Comment (zombieJ) [reply components/steps/style/index.tsx:33]: 等等，上面已经有 descriptionWidth 了，这个是不是不用的？

File: components/steps/style/index.tsx [hunk 5]
```diff
@@ -318,53 +306,52 @@ const genStepsStyle: GenerateStyle<StepsToken, CSSObject> = token => {
...
-    stepsNavActiveColor: token.colorPrimary,
-  });
+    const stepsToken = mergeToken<StepsToken>(token, {
+      // Steps variable default.less
+      processTailColor,
+      stepsNavArrowColor: new TinyColor('#000').setAlpha(0.25).toRgbString(), //  fade(@black, 25%),
+      stepsIconSize,
+      stepsIconCustomSize: stepsIconSize,
+      stepsIconCustomTop: 0,
+      stepsIconCustomFontSize: token.fontSizeIcon * 2,
```
Comment (zombieJ) [inline components/steps/style/index.tsx:323]: fontSizeHeading3

**Pull request merged**